<div id="navegacion" prev="../Tema_4/t4_funciones_02_pdi.html"
                     next="../Ejercicios_Tema_3/ej2.html"></div>

<div class="section-title">Sección 2.8</div>

# Ejercicios

## Ejercicio 1

Escribe el código Python que representa la siguiente expresión:

$$\frac{3}{4} + 2 * ( 5^4 - \frac{1}{2} )$$



In [ ]:
# Sol:
3 / 4 + 2 * (5**4 - 1/ 2)



1249.75

Escribe una función en Python llamada __valor_expresion__ que calcule el valor de la expresión:

$$\frac{3}{4} + 2 * ( x^4 - \frac{1}{2} )$$

para un cierto valor de $x$. La función tiene un único argumento posicional de tipo numérico y devuelve el resultado de aplicar la expresión anterior.

In [ ]:
# Sol
def valor_expresion(x):
      potencia = x**4
      res = 3/4 + 2 * (potencia - 1/2)
      return res

Ejecuta la función __valor_expresion__ para los valores 5 y 3.

__Nota:__

* `valor_expresion(5)` devuelve como resultado `1249.75`
* `valor_expresion(3)` devuelve como resultado `161.75`

In [ ]:
# Sol
result = valor_expresion(5)
print (result)
result = valor_expresion(3)
result

1249.75


161.75

## Ejercicio 2

Define una función llamada __raiz_cubica__ que devuelva el valor de $\sqrt[3]{x}$. Recuerda que  $\sqrt[3]{x}$ = ${x}^{\frac{1}{3}}$


__Nota:__

* `raiz_cubica(5)` devuelve como resultado `1.7099759466766968`

In [ ]:
# Sol:
def raiz_cubica(x):
  res = x**(1/3)
  return res

In [ ]:
result = raiz_cubica(5)
result

1.7099759466766968

## Ejercicio 3

Escribe una función llamada __pertenencia__ que reciba dos parámetros de entrada. El primer parámetro es una lista de elementos y el segundo es un elemento. La función devuelve el valor `True` si el elemento se encuentra en la lista. En otro caso devuelve `False`.

* Puedes usar la instrucción `if-else`
* Puedes usar la función `in` para saber si un elemento está en una secuencia

__Nota:__

* `pertenencia([1, 3, 6], 1)` devuelve el valor `True`
* `pertenencia([1, 3, 6], 5)` devuelve el valor `False`
* `pertenencia([1,'Hi', 6, 'Hola'] , 'Hola')` devuelve el valor `True`

In [ ]:
# Sol:
def pertenencia(x, y):
  res = y in x
  return res

In [ ]:
result = pertenencia([1, 3, 6], 1)
print (result)
result = pertenencia([1, 3, 6], 5)
print (result)
result = pertenencia([1,'Hi', 6, 'Hola'] , 'Hola')
result

True
False


True

## Ejercicio 4

La función `upper` del tipo string (`str`) genera una nueva cadena de caracteres, donde todos los caracteres son mayúsculas.
Ejecuta la siguente celda para ver el efecto de la función `upper` al aplicarlo sobre la cadena `s`:

In [ ]:
s = 'Hola, me llamo Iñigo Montoya, tú mataste a mi padre. ... Prepárate a morir'
s.upper()

'HOLA, ME LLAMO IÑIGO MONTOYA, TÚ MATASTE A MI PADRE. ... PREPÁRATE A MORIR'

Fíjate que `upper` no modifica la cadena (las cadenas son inmutables), sino que devuelve otra cadena diferente.

In [ ]:
s    # la cadena s no cambia de valor

'Hola, me llamo Iñigo Montoya, tú mataste a mi padre. ... Prepárate a morir'

Para guardar el resultado de `upper` usamos la asignación:

In [ ]:
t = s.upper()

Podemos  preguntar por el contenido de `t`:

In [ ]:
t

'HOLA, ME LLAMO IÑIGO MONTOYA, TÚ MATASTE A MI PADRE. ... PREPÁRATE A MORIR'

* Genera una cadena `t`  igual que  `s`  pero en minúsculas (función `lower`). Calcula la longitud de `t`.
* Ejecuta la función  `split(',')`. Fíjate en lo que devuelve esa función.
* Ejecuta la función  `title()`.
* Ejecuta la función  `replace(' ', '_')`. Fíjate en lo que devuelve esa función.
* Ejecuta `dir(str)` para listar las funciones disponibles para trabajar con cadenas.

In [ ]:
# Sol:
t = s.lower()
n = len(t)
print(n)
n = t.split(',')
print(n)
n = t.title()
print(n)
n = t.replace(' ', '_')
print(n)
dir(t)


74
['hola', ' me llamo iñigo montoya', ' tú mataste a mi padre. ... prepárate a morir']
Hola, Me Llamo Iñigo Montoya, Tú Mataste A Mi Padre. ... Prepárate A Morir
hola,_me_llamo_iñigo_montoya,_tú_mataste_a_mi_padre._..._prepárate_a_morir


['__add__',
 '__class__',
 '__contains__',
 '__delattr__',
 '__dir__',
 '__doc__',
 '__eq__',
 '__format__',
 '__ge__',
 '__getattribute__',
 '__getitem__',
 '__getnewargs__',
 '__getstate__',
 '__gt__',
 '__hash__',
 '__init__',
 '__init_subclass__',
 '__iter__',
 '__le__',
 '__len__',
 '__lt__',
 '__mod__',
 '__mul__',
 '__ne__',
 '__new__',
 '__reduce__',
 '__reduce_ex__',
 '__repr__',
 '__rmod__',
 '__rmul__',
 '__setattr__',
 '__sizeof__',
 '__str__',
 '__subclasshook__',
 'capitalize',
 'casefold',
 'center',
 'count',
 'encode',
 'endswith',
 'expandtabs',
 'find',
 'format',
 'format_map',
 'index',
 'isalnum',
 'isalpha',
 'isascii',
 'isdecimal',
 'isdigit',
 'isidentifier',
 'islower',
 'isnumeric',
 'isprintable',
 'isspace',
 'istitle',
 'isupper',
 'join',
 'ljust',
 'lower',
 'lstrip',
 'maketrans',
 'partition',
 'removeprefix',
 'removesuffix',
 'replace',
 'rfind',
 'rindex',
 'rjust',
 'rpartition',
 'rsplit',
 'rstrip',
 'split',
 'splitlines',
 'startswith',
 'stri

--------------------